Run forge tests with cairo steps explicitly set in tests

THenry14 opened the pull request:
Related to #2977

This PR stack aims to add support for new resource tracking (sierra-gas) and new gas representation (GasVector triplet instead of single l1 gas value).

-- (https://github.com/foundry-rs/starknet-foundry/pull/3049) Add sierra gas tracking
--> (This PR) Run already existing tests with CairoSteps set explicitly
-- (https://github.com/foundry-rs/starknet-foundry/pull/3051) Add support for sierra gas in `--detailed-resources`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3052) Add support for sierra gas in `--save-trace-data`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3053) Add sierra version assertion
-- (https://github.com/foundry-rs/starknet-foundry/pull/3054) Introduce new gas representation
-- (https://github.com/foundry-rs/starknet-foundry/pull/3055) Add gas tests for sierra-gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3056) Update docs with sierra-gas related changes

THenry14 committed:
Run forge tests with cairo steps explicitly set in tests

ksew1 reviewed (approved):
I am not sure about this change. Many tests do not care about the traced resource, so it seems unnecessary. Additionally, it suggests that `CairoSteps` is essential for this test to work.

THenry14 commented:
> I am not sure about this change. Many tests do not care about the traced resource, so it seems unnecessary. Additionally, it suggests that `CairoSteps` is essential for this test to work.

Some do, but mainly this just lays ground for https://github.com/foundry-rs/starknet-foundry/pull/3055, where you get different cost depending on tracked resource

THenry14 committed:
Adjust --detailed-resources to sierra-gas tracking (#3051)

Related to #2977

This PR stack aims to add support for new resource tracking (sierra-gas)
and new gas representation (GasVector triplet instead of single l1 gas
value).

-- (https://github.com/foundry-rs/starknet-foundry/pull/3049) Add sierra
gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3050) Run
already existing tests with CairoSteps set explicitly
-- (This PR) Add support for sierra gas in `--detailed-resources`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3052) Add
support for sierra gas in `--save-trace-data`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3053) Add sierra
version assertion
-- (https://github.com/foundry-rs/starknet-foundry/pull/3054) Introduce
new gas representation
-- (https://github.com/foundry-rs/starknet-foundry/pull/3055) Add gas
tests for sierra-gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3056) Update
docs with sierra-gas related changes
